React/wbooks/src/redux/auth/reducer.js

```diff
@@ -0,0 +1,112 @@
+import { authActions } from './actions';
+
+/* ------------- Auth reducer ------------- */
+const initialState = {
+  logState: {
+    logging: false,
+    loginFail: false,
+    logout: false,
+    signing: false,
+    signupfail: false
+  }
+};
+
+/* eslint-disable complexity */
+export function reducer(state = initialState, action) {
+  switch (action.type) {
+    case authActions.login: {
+      return {
+        ...state,
+        logState: {
+          logging: true,
+          loggedIn: false,
+          loginFail: false,
+          logout: false,
+          signing: false,
```

Comment:
I don't think you need to manipulate any signup state when performing a login action. Each action should change the state according to what they do, not necesarily changing every field in the current state. For example, you won't be doing a sign up when being in the login screen. So you'll always have:
```
{
  signing: false,
  signupfail: false
}
```

So there's no need to change those two fields in this switch case. Same goes for the others.